src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -265,7 +269,7 @@ public static function createConnection($dsn, array $options = [])
                 }
 
                 try {
-                    $redis = new $class(null, $hosts, $params['timeout'], $params['read_timeout'], (bool) $params['persistent'], $params['auth'] ?? '');
+                    $redis = new $class(null, $hosts, $params['timeout'], $params['read_timeout'], (bool) $params['persistent'], $params['auth'] ?? '', $params['ssl'] ?? null);
```

Comment:
I don't see any `ssl` argument in the source. Should be removed to me.
If it were to be confirmed as needed, there is already a $tls variable around that should be reused. We should not introduce a new param.

Reply:
The ssl value passed into RedisCluster needs to be null or an array. If its anything other than null, RedisCluster will try to use ssl.

https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L470
https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L373

In redis_cluster.c:470 the context value gets passed to redis_cluster_init. on redis_cluster.c:373 this gets passed into  redis_sock_set_stream_context.

The value passed to this last parameter of RedisCluster should be null, or an array of socket context options.
https://www.php.net/manual/en/context.php

Reply:
```yaml
services:
    app.cache.provider:
        class: \RedisCluster
        factory: ['Symfony\Component\Cache\Adapter\RedisTagAwareAdapter', 'createConnection']
        arguments:
            - '%env(resolve:REDIS_SCHEME)%:?host[%env(resolve:REDIS_HOST)%:%env(resolve:REDIS_PORT)%]'
            -
                class: \RedisCluster
                redis_cluster: true
                persistent: true
                # persistent_id: null
                lazy: true

                timeout: 60
                read_timeout: 1
                # retry_interval: 0

                ssl:
                    verify_peer: false
```

Reply:
When using predis, We are using the same configuration values.

https://packagist.org/packages/predis/predis

Reply:
Thanks for the insights. Looking at the source, it looks like `Redis::connect()` also has this ssl context argument. Can you pass the param there too? I didn't find it for RedisSentinel not RedisArray, did you?
Should we add `ssl` to `$defaultConnectionOptions` in order to hint that this exists?

Reply:
Check https://github.com/phpredis/phpredis/issues/1600#issuecomment-776109815 also!

Reply:
https://github.com/phpredis/phpredis/issues/1600

This is correct, ssl array should be an array of ssl socket context options.

https://www.php.net/manual/en/context.ssl.php

Reply:
Added to SSL. Should I document ssl config somewhere?

Reply:
Negative on ```RedisSentinel``` and ```RedisArray```.

Reply:
> Negative on RedisSentinel and RedisArray.

can you please create an issue about this on phpredis?

Reply:
Yep

Reply:
Do you want another issue filed on the consistency of the ->scan method?

Reply:
I'm not sure, it's done now.